pyOCD/gdbserver/gdbserver.py

```diff
@@ -363,11 +391,15 @@ def run(self):
                         if self.persist:
                             break
                         else:
-                            self._cleanup()
+                            self.shutdown_event.set()
                             return
 
                 self.lock.release()
 
+            except Exception as e:
```

Comment:
I think exceptions _should_ be handled at the command level. These new exception handlers were added just to ensure we don't exit the connection loop while gdb still thinks we're still connected, or let an unhandled exception terminate the thread.

If wanted to return a generic error response, we could add an exception handler in `handleMsg()`. In these two exception handlers in `run()` and `_run_connection()`, we don't necessarily know that gdb is waiting for a response.